esti/catalog_export_test.go

```diff
@@ -0,0 +1,418 @@
+
+	s3Client, err := testutil.SetupTestS3Client("https://s3.amazonaws.com", testData.AccessKeyId, testData.SecretAccessKey)
+
+	require.NoError(t, err, "failed creating s3 client")
+
+	// we need to retry for the result
+	var symlinkLocations []string
+
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = 5 * time.Second
+	bo.MaxElapsedTime = 30 * time.Second
+	listS3Func := func() error {
+		listResp, err := s3Client.ListObjectsV2(ctx, &s3.ListObjectsV2Input{
+			Bucket: aws.String(storageURL.Host),
+			Prefix: aws.String(storageURL.Path[1:]), // trim leading slash
+		})
+		require.NoErrorf(t, err, "listing symlink files in storage: %s", symlinksPrefix)
+		if len(listResp.Contents) == 0 {
+			return fmt.Errorf("no objects found")
+		}
+		for _, f := range listResp.Contents {
+			symlinkLocations = append(symlinkLocations, aws.ToString(f.Key))
+		}
+		return nil
+	}
+
+	err = backoff.Retry(listS3Func, bo)
+
+	require.NoErrorf(t, err, "failed listing symlink files %s", symlinksPrefix)
+	require.NotEmptyf(t, symlinkLocations, "no symlink files found in blockstore: %s", symlinksPrefix)
+
+	// get the symlink files and compare their physical address to lakefs result
+	storagePhysicalAddrs := map[string]bool{}
+	for _, symlinkFileKey := range symlinkLocations {
+		objRes, err := s3Client.GetObject(ctx, &s3.GetObjectInput{
+			Bucket: aws.String(storageURL.Host),
+			Key:    aws.String(symlinkFileKey),
+		})
+		require.NoErrorf(t, err, "getting symlink file content bucket=%s key=%s", storageURL.Host, symlinkFileKey)
+		defer objRes.Body.Close()
+		body, err := io.ReadAll(objRes.Body)
+		require.NoError(t, err, "fail reading object data")
+		for _, addr := range strings.Split(string(body), "\n") {
+			if addr != "" { // split returns last \n as empty string
+				storagePhysicalAddrs[addr] = true
+			}
+		}
+	}
+
+	lakeFSObjs, err := client.ListObjectsWithResponse(ctx, repo, commit.Id, &apigen.ListObjectsParams{
+		Prefix: apiutil.Ptr(apigen.PaginationPrefix(testData.TableSpec.Path)),
+	})
+
+	require.NoError(t, err, "failed listing lakefs objects")
+
+	// test that all lakeFS entries are exported and represented correctly in symlink files
+	lakefsPhysicalAddrs := map[string]bool{}
+	for _, entry := range lakeFSObjs.JSON200.Results {
+		if !strings.Contains(entry.Path, "_hidden") {
+			lakefsPhysicalAddrs[entry.PhysicalAddress] = true
+		}
+	}
+
+	require.Equal(t, lakefsPhysicalAddrs, storagePhysicalAddrs, "mismatch between lakefs exported objects in symlink files")
+
+	return commit.Id, symlinksPrefix
+}
+
+// TestAWSCatalogExport will verify that symlinks are exported correcrtly and then in a sequential test verify that the glue exporter works well.
+// The setup in this test includes:
+// Symlinks export: lua script, table in _lakefs_tables, action file, mock table data in CSV form
+// Glue export: lua script, table in _lakefs_tables, action file
+func TestAWSCatalogExport(t *testing.T) {
+	requireBlockstoreType(t, block.BlockstoreTypeS3)
```

Comment:
I am not sure that this is the ideal thing to use.  We are interested in running on AWS, not in running of S3.  Suppose we were running on MinIO: presumably Glue would not work there.

Reply:
Done! Thanks!